packages/components/ContextualMenu/src/SubmenuItem.tsx

```diff
@@ -110,10 +110,11 @@ export const SubmenuItem = compose<SubmenuItemType>({
       icon: !!icon,
     };
 
-    const showWithArrowKey = React.useCallback(
+    const showSubmenuOnKeyDown = React.useCallback(
       (e: any) => {
-        const arrowKey = I18nManager.isRTL ? 'ArrowLeft' : 'ArrowRight';
-        if (e.nativeEvent.key === arrowKey) {
+        const rtlOpenArrowKey = I18nManager.isRTL ? 'ArrowLeft' : 'ArrowRight';
+
+        if (e.nativeEvent.key === rtlOpenArrowKey || e.nativeEvent.key === ' ' || e.nativeEvent.key === 'Enter') {
```

Comment:
nit, `rtlOpenArrowKey` sounds like the key specifically for RTL, rather than the locale aware key.